### PR TITLE
Add better documentation

### LINKED ISSUE #193: Documentation: improvements on the current state and planned features
**Describe the enhancement you'd like to see**

* what features are supported and what are planned using clear enumeration (table?). High level Roadmap would be nice.
* more better documentation on how KGW is different from other gateway, what problems it solves.

### PR DESCRIPTION
**Pull request by Tarick**
This PR...

## Changes

Improves the documentation to be more clear as well as adding the roadmap.

## Fixes

Closes #193

## Checklist

- [ ] tested locally
- [ ] added new dependencies
- [x] updated the docs
- [ ] added a test

### CODE REVIEW

File: docs/arch.md [hunk 1]
```diff
@@ -1,2 +1,13 @@
 # Architecture
-![kusk-gateway arch diagram](img/arch.png)
\ No newline at end of file
+
+Under the hood, Kusk Gateway consists of an Operator (Kusk Gateway Manager) and created by it K8s resources.
```
Comment (kylehodgetts) [inline docs/arch.md:3]: `created by it K8s resources` not really sure what this means?

Do you mean the operator is created by K8s resources?

Comment (Tarick) [reply docs/arch.md:3]: created by the operator k8s resources

Comment (Tarick) [reply docs/arch.md:3]: suggestions for the better phrasing are welcome :)

Comment (kylehodgetts) [reply docs/arch.md:3]:
```suggestion
Under the hood, Kusk Gateway consists of an Operator (Kusk Gateway Manager), created by K8s resources.
```

File: docs/arch.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Architecture
 
-Under the hood, Kusk Gateway consists of an Operator (Kusk Gateway Manager) and created by it K8s resources.
+Under the hood, Kusk Gateway consists of an Operator (Kusk Gateway Manager) and k8s resources, created by that operator.
```
Comment (kylehodgetts) [inline docs/arch.md:3]:
```suggestion
Under the hood, Kusk Gateway consists of an Operator (Kusk Gateway Manager) and k8s resources, created and managed by the operator. e.g. APIs, StaticRoutes and EnvoyFleets
```

**Review by kylehodgetts — APPROVED** [review]

**Pull request merged**